### ISSUE TITLE
npm install -g ethereumjs-testrpc

### ISSUE THREAD

**Issue by MannieIW** [ConsenSys-archive/ganache#1206]
I have node.js v10.19.0. And have Ubuntu in VirtualBox. I tried to run 'npm install -g ethereumjs-testrpc', but it returned errors.
Please how do I resolve this? I am trying to set up an IDE for Solidity.
Kindly see below:

mannieiw@mannieiw-VirtualBox:~$ node -v
v10.19.0
mannieiw@mannieiw-VirtualBox:~$ npm install -g ethereumjs-testrpc
npm WARN deprecated ethereumjs-testrpc@6.0.3: ethereumjs-testrpc has been renamed to ganache-cli, please use this package from now on.
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN checkPermissions Missing write access to /usr/local/lib
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/ethereumjs-testrpc/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/ethereumjs-testrpc/node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code EACCES
npm ERR! syscall access
npm ERR! path /usr/local/lib
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, access '/usr/local/lib'
npm ERR!  { [Error: EACCES: permission denied, access '/usr/local/lib']
npm ERR!   stack:
npm ERR!    'Error: EACCES: permission denied, access \'/usr/local/lib\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/usr/local/lib' }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mannieiw/.npm/_logs/2021-09-19T13_32_26_206Z-debug.log
mannieiw@mannieiw-VirtualBox:~$

**Comment by davidmurdoch** [comment]
Why are you trying to install `ethereumjs-testrpc` instead of `ganache-cli`?

**Comment by MannieIW** [comment]
Hi David,

Thank you for your response.

I am running an online training course on "Certified Solidity Developer",
organized by Blockchain Council. As part of setting up my Solidity
Integrated Development Environment (IDE), the online instructor asked to
run "npm install -g ethereumjs-testrpc". And the instructor proceeded to
demonstrate it at the blockchain council website and it worked well. The
instructor also said that if one installed node.js (+ npm) the 'npm install
-g ethereumjs-testrpc' should easily run successfully..
To confirm that I had properly downloaded node.js (+ npm), I ran 'node -v;
and 'npm -v' and it turned up their version numbers, v14.17.6 and v6.14.15
respectively.
But when I ran  'npm install -g ethereumjs-testrpc', it turned up
errors, which was the information that I sent to you. I am using Ubuntu in
VirtualBox on Windows 10.

Googling testrpc, amongst others, i noticed that it also turned up -
ethereumjs-testrpc
has been renamed to ganache-cli, please use this package from now on.
Clearly, since ganache-cli is the way to go, can you please advise how I
may proceed?

Thank you.

Best regards,

Emmanuel Iwerebon
Mobile: 469-315-9981.

**Comment by davidmurdoch** [comment]
That must be an old course, as testrpc was renamed to ganache-cli about 4 years ago. Install ganache-cli by running `npm install ganache-cli -g` instead. Let me know if you still have issues.

**Comment by MannieIW** [comment]
Hi David,

Thank you for your email.
I just ran *npm install ganache-cli -g*, kindly see below the result.

***@***.***:~$ node -v
v14.17.6
***@***.***:~$ npm -v
6.14.15
***@***.***:~$  npm install ganache-cli -g
npm WARN checkPermissions Missing write access to /usr/lib/node_modules
npm ERR! code EACCES
npm ERR! syscall access
npm ERR! path /usr/lib/node_modules
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, access '/usr/lib/node_modules'
npm ERR!  [Error: EACCES: permission denied, access
'/usr/lib/node_modules'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/usr/lib/node_modules'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file
as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please
double-check the
npm ERR! permissions of the file and its containing directories, or try
running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mannieiw/.npm/_logs/2021-09-21T23_22_40_518Z-debug.log
***@***.***:~$

Awaiting further instructions from you, please.

Thank you.

Best regards,

Emmanuel Iwerebon
469-315-9981.

**Comment by davidmurdoch** [comment]
If you've previously tried installing an npm module with the `-g` or `--global` flag along with `sudo` your npm directory permissions are likely misconfigured. See npm's help article on this here: https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally

**Comment by MannieIW** [comment]
Thank you for your email.
Will review and revert shortly.

Best regards,

Emmanuel Iwerebon
Mobile: 469-315-9981.

**Comment by MannieIW** [comment]
Hi David,

Your earlier email below refers.

I have tried to follow npm's help article but am still missing something.
I am honestly not sure if I did the right thing, but these are the steps
that I took.

Clicking on the help article url, it took me to coreybutler/nvm-windows.
I clicked on 'Code'
Then I clicked on Download ZIP
Then I extracted the ZIP file.
It displayed a folder: nvm-windows-master. I clicked on this folder
which then displayed from .github to SUPPORT.md

And it is not clear to me how to proceed from here.

Am very deeply sorry to be bothering you, but I really do need help with
this.
I am pretty much interested in learning the language - Solidity.

Thank you.

Best regards,

Emmanuel Iwerebon
Mobile: 469-315-9981.
